config.go

```diff
@@ -143,11 +145,26 @@ func getEffectiveConfig(configRoot, owner, repo string) (*effectiveConfig, error
 }
 
 func buildGlobalEffectiveConfig(global *config) effectiveConfig {
+	origServerID := firstNonEmptyStringPtr(global.GrimServerID, global.GrimQueueName, &defaultGrimQueueName)
+	truncatedServerID := origServerID
+	truncateID := ""
+
+	if len(truncatedServerID) > 15 {
+		truncatedServerID = fmt.Sprintf("%s", origServerID[:15])
+
+		truncateID = "GrimServerID"
+		if !stringPtrNotEmpty(global.GrimServerID) {
+			truncateID = "GrimQueueName"
+		}
+	}
+
```

Comment:
Whenever we get an effectiveConfig, we are reading it from the file. If we go with this approach, we would have to call performCorrections every time we use an effectiveConfig.